airflow/plugins/operators/scrape_state_geoportal.py

```diff
@@ -0,0 +1,206 @@
+import gzip
+import logging
+import os
+from typing import ClassVar, List
+
+import pandas as pd  # type: ignore
+import pendulum
+import requests
+from calitp_data_infra.storage import PartitionedGCSArtifact, get_fs  # type: ignore
+from pydantic import HttpUrl, parse_obj_as
+
+from airflow.models import BaseOperator  # type: ignore
+
+API_BUCKET = os.environ["CALITP_BUCKET__STATE_GEOPORTAL_DATA_PRODUCTS"]
+
+
+class StateGeoportalAPIExtract(PartitionedGCSArtifact):
+    bucket: ClassVar[str]
+    execution_ts: pendulum.DateTime = pendulum.now()
+    dt: pendulum.Date = execution_ts.date()
+    partition_names: ClassVar[List[str]] = ["dt", "execution_ts"]
+    root_url: str
+    service: str
+    product: str
+    where: str
+    outFields: str
+    f: str
+    resultRecordCount: int
```

Comment:
**suggestion:** This is mirroring the parameters of an ArcGIS Online service request, right? Could we add some comments for each of these options for folks that are familiar with AGO? Something like:

```suggestion
    bucket: ClassVar[str]
    execution_ts: pendulum.DateTime = pendulum.now()
    dt: pendulum.Date = execution_ts.date()
    partition_names: ClassVar[List[str]] = ["dt", "execution_ts"]
    
    # The name to be used in the data warehouse to refer to the data
    # product.
    product: str
    
    # The root of the ArcGIS services. As of Nov 2024, this should
    # be "https://caltrans-gis.dot.ca.gov/arcgis/rest/services/".
    root_url: str
    
    # The name of the service being requested. In the feature service's
    # URL, this will be everything between the root and "/FeatureServer".
    # Don't include a leading or trailing slash.
    service: str
    
    # The layer to query. This will usually be "0", so that is the 
    # default.
    layer: str = "0"
    
    # The query filter. By default, all rows will be returned from the
    # service. Refer to the ArcGIS documentation for syntax:
    # https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-layer/#request-parameters
    where: str = "1=1"
    
    # A comma-separated list of fields to include in the results. Use
    # "*" (the default) to include all fields.
    outFields: str = "*"
    
    # The number of records to request for each API call (the operator
    # will request all data from the layer in batches of this size).
    resultRecordCount: int
```

_(NOTE that I removed `f` and added `layer` in the suggestion above. I also moved `product` above the AGO-specific options. I'm not sure that the descirption that I provided for that parameter is correct.)_